### DISCUSSION TITLE
cargo-watch package in Zed's source code

### DISCUSSION THREAD

**Discussion by faniabdo99** [zed-industries/zed#10826] [Show and tell]
Hi All,
I was playing around with Zed's codebase and figured out something quite nice, when I built Zed on my Macbook Pro I noticed something might be annoying in the development process, there is no hot reload!

So I found a package called `cargo-watch` and installed it, worked like a charm.

I did create a branch and wanted to open a PR but I don't have the right permissions and I am not really sure how useful this is to the development team, so I am just gonna leave the code right here for anyone who can use it

```
cargo install cargo-watch
cargo watch -x run
```

With the above, every time you save a file it will rebuild the application and open it, this behavior can be adjusted by configuring `cargo-watch` the way you need, here are `cargo-watch` docs:
https://github.com/watchexec/cargo-watch

Thank you Zed team! you are doing awesome work out there!

**Comment by SomeoneToIgnore** [comment]
Just to note, if someone looks for less automated process,
Zed also provides a `cargo run` task now automatically for Rust projects:

<img width="553" alt="image" src="https://github.com/zed-industries/zed/assets/2690773/0638892d-2178-4d68-ac8f-14df91346d39">

and that one in Zed could be rerun with a single shortcut after edits.

**Reply by faniabdo99** [reply]
Yup, this is the default and can be found in the build documentation 👍
